kad/kad.go

```diff
@@ -28,3 +28,40 @@ type Key[K any] interface {
 	// and 0 if both keys are equal.
 	Compare(other K) int
 }
+
+// RoutingTable is the interface all Kademlia Routing Tables types support.
+type RoutingTable[K Key[K]] interface {
+	// Self returns the local node's Kademlia key
+	Self() K
+
+	// AddPeer tries to add a peer to the routing table
+	AddPeer(NodeID[K]) (bool, error)
+
+	// RemoveKey tries to remove a peer identified by its Kademlia key from the
+	// routing table
+	RemoveKey(K) (NodeID[K], error)
```

Comment:
We say `AddNode` but `RemoveKey` instead of `RemoveNode`.

Generally, we could solely operate on `NodeID[K]` in this interface. Not sure if this'll be too restrictive.

Reply:
It's because the Key can be derived from the Node but not the other way round.

We could have

`
AddNode(key, node)
RemoveNode(key)
`

but since we can get the key from the node using node.Key() we don't include it in the AddNode signature. I asked the same question of Gui when I was first looking into it.

Reply:
We could also remove the `error` from the signature because it seems that we only used it if the keys didn't match.

Reply:
> It's because the Key can be derived from the Node but not the other way round.

This comment is related to the `AddNode` method right? I was asking if should operate on `NodeID` in all methods. If we didn't need a `NodeID` in `AddNode` I'd say it's better to only use `K` everywhere, but since we need `NodeID[K]` in `AddNode` we could also add it to `RemoveNode` and `NearestNodes`.

Not sure if this is too restrictive though. No strong opinion here, just wanted to hear your thoughts.

Reply:
adjusted.